inc/class-wpseo-utils.php

```diff
@@ -759,6 +759,39 @@ public static function is_development_mode() {
 	}
 
 	/**
+	 * Retrieve home URL with proper trailing slash.
+	 *
+	 * @param string      $path   Path relative to home URL.
+	 * @param string|null $scheme Scheme to apply.
+	 *
+	 * @return string Home URL with optional path, appropriately slashed if not.
+	 */
+	public static function home_url( $path = '', $scheme = null ) {
+
+		$home_url = home_url( $path, $scheme );
+
+		if ( ! empty( $path ) ) {
+			return $home_url;
+		}
+
+		$home_path = parse_url( $home_url, PHP_URL_PATH );
+
+		if ( '/' === $home_path ) { // Home at site root, already slashed.
+			return $home_url;
+		}
+
+		if ( is_null( $home_path ) ) { // Home at site root, always slash.
+			return trailingslashit( $home_url );
+		}
+
+		if ( is_string( $home_path ) ) { // Home in subdirectory, slash if permalink structure has slash.
+			return user_trailingslashit( $home_url );
+		}
+
+		return $home_url;
```

Comment:
I don't think it is a good spot, no access to context. For changing it overall there are core filters. For changing URL in plugin there are specific filters by function (e.g. `wpseo_canonical` for canonical and so on).